hnn_core/parallel_backends.py

```diff
@@ -351,8 +393,23 @@ def simulate(self, net):
         self.sel.register(pipe_stderr_r, selectors.EVENT_READ,
                           self._read_stdout)
 
+        data_len = 0
+        completed = False
+        timeout = 0
         # loop while the process is running
-        while proc.poll() is None:
+        while True:
```

Comment:
A clean break is when: (1) the child process has sent the completion signal and (2) the child process has exited (i.e. `proc.poll() is not None`